### PR TITLE
Add callback logic for outgoing invoice

### PR DESCRIPTION
**Pull request by joshr4**

### CODE REVIEW

File: library/withdraw.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    .update({
+      success: status === PaymentStatus.Completed,
+      is_pending: false,
+      updated_at: db.knex.fn.now(),
+      fee_msat: fee,
+      preimage: preimage,
+    })
+    .where({ id: transactionId })
+    .then(() => {
+      // Increment user balance and unlock user
```
Comment (blastshielddown) [inline library/withdraw.ts:32]: nit
```suggestion
      // Decrement user balance and unlock user
```

File: library/withdraw.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      return trx("user")
+        .decrement({
```
Comment (blastshielddown) [inline library/withdraw.ts:34]: What if the outgoing payment failed? Where does that check happen?

File: services/accounting/routes/withdraw.ts [hunk 3]
```diff
@@ -24,9 +24,9 @@ const limiter = rateLimit({
 router.get("/", isAuthorized, sendController.getWithdraw);
 router.post(
   "/",
-  isAuthorized,
-  limiter,
-  isZbdRegion,
+  // isAuthorized,
+  // limiter,
+  // isZbdRegion,
```
Comment (blastshielddown) [inline services/accounting/routes/withdraw.ts:29]: Were these commented out for testing?

File: library/withdraw.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      fee_msat: fee,
+      preimage: preimage,
+    })
+    .where({ id: transactionId })
+    .then(() => {
+      if (status === PaymentStatus.Completed) {
+        // Increment user balance and unlock user
+        return trx("user")
+          .decrement({
+            msat_balance: msatAmount,
```
Comment (blastshielddown) [inline library/withdraw.ts:36]: Should include the fee on this

Comment (joshr4) [reply library/withdraw.ts:36]: oh good catch

**Review by blastshielddown — APPROVED** [review]

**Pull request merged**